src/modules/returns/lib/jobs/start-upload.js

```diff
@@ -50,22 +50,21 @@ const validateS3Object = async (evt, s3Object) => {
  */
 const handleReturnsUploadStart = async job => {
   const { eventId } = job.data;
-  const evt = await event.load(eventId);
+
+  const event = await eventsService.findOne(eventId);
 
   try {
-    const s3Object = await returnsUpload.getReturnsS3Object(eventId, evt.subtype);
+    const s3Object = await returnsUpload.getReturnsS3Object(eventId, event.subtype);
 
     // Pass parsed xml or csv doc to the validation function
     // returns true if the validation passes
     // returns an array of objects containing error messages and lines
-    await validateS3Object(evt, s3Object);
-
-    return job.done();
+    await validateS3Object(event, s3Object);
```

Comment:
I think this is OK, more recent versions of PG boss have moved from the callback style (job.done()) to a promise style, where the job handler returns a promise which resolves (job done OK) or rejects (job failed).
I believe using await here ensures any error thrown is caught by the try/catch, if I had done:
```
return validateS3Object(event, s3Object);
```
this would not have been the case

Reply:
Makes sense, thanks James :)